decode.go

```diff
@@ -193,7 +198,14 @@ func (d *decoder) decodeSignedInt(wiretype int, v uint64) (int64, error) {
 }
 
 func (de *decoder) putvalue(wiretype int, val reflect.Value,
-	v uint64, vb []byte) error {
+	v uint64, vb []byte) (err error) {
```

Comment:
I'm confused why this needs a recover; I would expect there should be a recover only in the public function (i.e. DecodeWithConstructors) and that one recover would catch all panics from inside of the library.

Reply:
oh my bad, I thought we had to build the whole recover pipeline for it to get passed to the public function. I'll remove it.